rfd/xxxx-automatic-azure-server-discovery.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: Andrew Burke (andrew.burke@goteleport.com)
+state: draft
+---
+
+# RFD X - Automatic discovery of Azure servers
+
+## What
+
+Teleport SSH services will be able to automatically discover Azure virtual machine
+instances. See [RFD 57](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md)
+(Automatic discovery and enrollment of AWS servers) for the same feature implemented
+for AWS.
+
+Note: unlike RFD 57, this RFD covers discovery only. Azure node joining will be added
+later.
+
+## Why
+
+RFD 57 replaced the process of manually installing Teleport on EC2 servers, which
+could be slow for large numbers of instances. This RFD replaces the process on Azure
+as well.
+
+## Discovery
+
+Azure discovery will be handled by the new [discovery service](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md#discovery) described in RFD 57.
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  azure:
+    - types: ["vm"]
+      regions: ["westcentralus"]
+      tags:
+        "teleport": "yes"
+```
+
+The Teleport SSH service will need a [service principal](https://learn.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest) with a role that includes the `Microsoft.Compute/virtualMachines/read`
+permission to list virtual machines via the Go Azure SDK.
+
+As with AWS database discover and EC2 discover, new Azure nodes will be discovered
```

Comment:
How do you plan to add them? 
Currently, EC2 works with SSM, and nodes are only added to the cluster once they register themselves (after discovery service calls SSM to install Teleport on them).

Reply:
I imagine it will be mostly the same Azure-equivalent process, but I think those details would fit better in the eventual registration RFD.